### ISSUE TITLE
Amplitude dashboard with macOS filter return iPadOS statistics

### ISSUE THREAD

**Issue by aureliendelrue** [amplitude/Amplitude-Swift#132]
On the Amplitude dashboard, if I want statistics about macOS version, the dashboard retrieve iPadOS data.
<img width="2733" alt="Capture d’écran 2024-03-08 à 08 46 13" src="https://github.com/amplitude/Amplitude-Swift/assets/25080393/306a7979-2248-484d-ac1a-4787086c86bc">

**Comment by izaaz** [comment]
@aureliendelrue - We fetch the OS information from `UIDevice.current.systemName`. This is most likely because you've published an iOS/iPad app with macos support. On M1s, Apple lets users download the ipad version of the apps as a desktop app.

**Comment by aureliendelrue** [comment]
@izaaz You're right if you have the OS information from `UIDevice.current.systemName` but maybe you could use [this](https://mandarapte.com/apple/programmatically-get-ios-ipados-macos-catalyst-operating-system-name-and-version-and-a-model-identifier-of-iphone-ipad-mac-devices/)

**Comment by izaaz** [comment]
Thank you for the suggestion @aureliendelrue. I'll keep this issue open so we can get more feedback from others who run into this issue and decide what can be done next.

**Comment by aureliendelrue** [comment]
Hi ! Any news for the bug ?

**Comment by emartinson** [comment]
@izaaz Hi! Thank you for reply! But this is definitely a bug. From real-world environment perspective the actual system is MacOS. We'll be kindly appreciate if you add check for `ProcessInfo.processInfo.isiOSAppOnMac` and `isMacCatalystApp` which are both MacOS.

```swift
    #if os(iOS)
                    if ProcessInfo.processInfo.isiOSAppOnMac {
                        log.info("[AU] Hello from macOS")
                    } else {
                        log.info("[AU] Hello from iOS")
                    }
            #elseif os(macOS)
                    log.info("[AU] Hello from macOS")
            #elseif targetEnvironment(macCatalyst)
                    log.info("[AU] Hello from Catalyst")
            #endif
```

**Comment by emartinson** [comment]
@aureliendelrue the only thing I figured out as workaround is write an enrighment plugin:

```swift
final class PlatformEnrichmentPlugin: EnrichmentPlugin {

    override func execute(event: BaseEvent?) -> BaseEvent? {
        guard let event else { return nil }
        event.userProperties = event.userProperties ?? [:]
        event.eventProperties = event.eventProperties ?? [:]
        event.groupProperties = event.groupProperties ?? [:]

        func forceMacOS() {
            event.platform = "macOS"
            event.deviceModel = "macOS_Unknown_" + (event.deviceModel ?? "")
            event.osVersion  = ProcessInfo.processInfo.operatingSystemVersionString
            event.osName = "macOS_" + (event.osName ?? "")
        }

        #if os(iOS)
        if ProcessInfo.processInfo.isiOSAppOnMac {
            forceMacOS()
        }
        #elseif os(macOS) || targetEnvironment(macCatalyst)
        forceMacOS()
        #endif
        return event
    }
}
```

and then add it with

```swift
aimplitude.add(plugin: PlatformEnrichmentPlugin())
```

This will definitely give us correct platform. But detecting environment is tricky part. I was able to get MacOS version from ProcessInfo, but OS Name /  Model - is quite difficult to figure out (I found several approaches with touching system plists, but haven't tested them yet). If anyone can offer a working solution - please post it here.

Above plugin will give you these `Platform`, `OS` and `Device Family`:
<img width="600" alt="image" src="https://github.com/user-attachments/assets/8f6eba09-e654-4fa4-9830-a9f54960e108">
